typed-racket-lib/typed-racket/static-contracts/combinators/structural.rkt

```diff
@@ -157,6 +157,7 @@
   ((listof/sc (#:covariant)) listof #:flat)
   ((cons/sc (#:covariant) (#:covariant)) cons/c #:flat)
   ((set/sc (#:covariant #:chaperone)) set/c #:flat)
+  ((struct-property/sc (#:invariant)) struct-type-property/c #:flat)
   ((immutable-vectorof/sc (#:covariant)) immutable-vectorof/c #:flat)
```

Comment:
Is this always a flat contract? What if it's a function type within the struct-property? Also, off the top of my head I can't decide if `#:invariant` makes sense and what the implications of that would be vs covariant, so if you haven't already maybe double check that is fine.

Reply:
```
> (flat-contract? (struct-type-property/c exact-nonnegative-integer?))
#f
> (chaperone-contract? (struct-type-property/c exact-nonnegative-integer?))
#f
> (impersonator-contract? (struct-type-property/c exact-nonnegative-integer?))
#t
```